pennylane/wires.py

```diff
@@ -0,0 +1,369 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the :class:`Wires` class, which takes care of wire bookkeeping.
+"""
+from collections import Sequence, Iterable
+import numpy as np
+from numbers import Number
+
+
+class WireError(Exception):
+    """Exception raised by a :class:`~.pennylane.wires.Wire` object when it is unable to process wires.
+    """
+
+
+def _process(wires):
+    """Converts the input to a tuple of numbers or strings."""
+
+    if isinstance(wires, Wires):
+        # if input is already a Wires object, just copy its wire tuple
+        return wires.wire_tuple
+
+    elif isinstance(wires, Iterable) and all(isinstance(w, Wires) for w in wires):
+        # if the elements are themselves Wires objects, merge them to a new one
+        wires = [w for wires_ in wires for w in wires_.tolist()]
+        return tuple(wires)
+
+    elif isinstance(wires, Iterable) and all(isinstance(w, str) or isinstance(w, Number) for w in wires):
+        # if the elements are strings or numbers, turn iterable into tuple
+        return tuple(wires)
```

Comment:
Minor nitpick: this could be merged with the above if statement. E.g.,

```python
elif isinstance(wires, Iterable):
    if all(isinstance(w, str) or isinstance(w, Number) for w in wires):
        # if the elements are strings or numbers, turn iterable into tuple
        return tuple(wires)

    elif all(isinstance(w, Wires) for w in wires):
        wires = [w for wires_ in wires for w in wires_.tolist()]
        return tuple(wires)
```

Reply:
funny enough, your code is a bit different: I want Iterables that are not Numbers, Wires or strings to throw the error. In your code that is not possible